### ISSUE TITLE
Endar Spire Command Module - Trask unideal coordinate/position jumping in between opening dialogue

### ISSUE THREAD

**Issue by ebmar** [KOTORCommunityPatches/K1_Community_Patch#86]
It always strikes me when Trask `end_trask.utc` do that _jumper_ stuff on the opening [Tutorial] dialogue `end_trask01.dlg` in (Endar Spire) Command Module `end_m01aa`. Attached a screenshot for clear illustration:

![Trask_Jumping_Position](https://user-images.githubusercontent.com/50026566/56866883-e83b6500-6a08-11e9-9d9d-01c16abcfee6.jpg)

Perhaps to remove the jumping sequence so he'd stand where he was would be better for immersion/gameplay purpose?

**Comment by DarthParametric** [comment]
There's no "jump" as such I don't think. It's a consequence of destroying the cutscene Trask and spawning the party Trask. I assume it's because the party members are always stationed behind the player, so naturally party Trask spawns in behind the player, causing the apparent jump from one side to the other. 

I'm not sure if you can dictate a spawn location when adding a party member via the party selection panel. Thoughts @JCarter426 ?

**Comment by JCarter426** [comment]
Hmm, I don't know if there's any way to do it that would be seamless. Theoretically you could edit Trask's OnSpawn script to put him a specific location, but I'd expect him to still be briefly visible in the original position.
Even when you add an existing creature to the party rather than destroying & spawning a new one, they do this sliding sort of thing that looks odd.

**Comment by DarthParametric** [comment]
Could we perhaps spawn the party Trask in ahead of time, hidden off-screen, and then jump him into position while the party selection panel is masking the screen?

**Comment by ebmar** [comment]
> It's a consequence of destroying the cutscene Trask and spawning the party Trask. I assume it's because the party members are always stationed behind the player...

Yeah, I partially thought about that too. Perhaps it's quite similar with how Canderous joins us on Davik's Estate. 

> ...edit Trask's OnSpawn script to put him a specific location, but I'd expect him to still be briefly visible in the original position.

Sure, shouldn't break another thing while fixing the other one.

**Comment by JCarter426** [comment]
![swkotor 2019-04-29 13-35-41-77](https://user-images.githubusercontent.com/46898647/56915144-ccc17000-6a83-11e9-92d0-8ba635f63869.png)
I managed to teleport him via the party selection's exit script, but he faces the wrong way. Not sure why, changing the orientation of the location didn't make a difference.

**Comment by DarthParametric** [comment]
Facing issues in conversations. Welcome to my world.

**Comment by ebmar** [comment]
> I managed to teleport him via the party selection's exit script, but he faces the wrong way.

Dang - that was close!

**Comment by DarthParametric** [comment]
I assume you tried a facing command to no effect? You could try forcing an animation afterwards. That's why I have had to walk the party forwards in certain scenes. They just refuse to obey facing commands otherwise.

**Comment by JCarter426** [comment]
Yeah, he ignores the facing command because the conversation has started by then, via ExecuteScript() of the original code. And changing the order of that results in the conversation not starting, for some reason.
I tried to recreate the original code to do things in my own order, but once I messed with it, the conversation wouldn't start, or he wouldn't jump to his location, or both.

**Comment by DarthParametric** [comment]
Can you force an animation on the first node of the DLG?

**Comment by JCarter426** [comment]
Tried that, and with speaker and listener set. I think the issue is that he's teleported after the conversation starts, so he's stuck with the facing angle of where he spawned. Only way to fix that would be to start the conversation after he is correctly in place, but he isn't cooperating with that.

**Comment by DarthParametric** [comment]
The solution is to offload the commands elsewhere instead of running them in the script that fires along with the party screen:

https://youtu.be/lGNT1FT9aeg

Could be that same problem I was having with the murder mystery corpse, where adding delays meant the owner was destroyed before the script completed.

**Comment by DarthParametric** [comment]
It seems that Trask will still teleport under certain conditions. Seems like the player will need to be jumped when re-initiating conversation after getting their gear.

**Comment by Salk73** [comment]
If we find a permanent solution to this, it should be applied also to Mission joining the party in Taris Undercity too.

**Comment by Salk73** [comment]
While testing this fix I have encountered (twice out of 10 times so far) a critical issue which I don't think is related to this fix but never encountered before.

What happens is that sometimes Trask won't make it through the door and when the conversation with the player starts, he is unreachable on the other side.

**Comment by DarthParametric** [comment]
The opening cutscene? While the same DLG, that is a wholly different set of scripts. You'll need to start a separate issue for it.

**Comment by Salk73** [comment]
Yes, exactly that one.

I'll open a separate issue.
